flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/command/ShowJarsOperation.java

```diff
@@ -34,10 +48,53 @@ public String asSummaryString() {
 
     @Override
     public TableResultInternal execute(Context ctx) {
-        String[] jars =
-                ctx.getResourceManager().getResources().keySet().stream()
+        ResourceManager resourceManager = ctx.getResourceManager();
+        FunctionCatalog functionCatalog = ctx.getFunctionCatalog();
+        CatalogManager catalogManager = ctx.getCatalogManager();
+        Set<String> jars = getJars(resourceManager, functionCatalog, catalogManager);
+
+        return buildStringArrayResult("jars", jars.toArray(new String[0]));
+    }
+
+    public static Set<String> getJars(
+            ResourceManager resourceManager,
+            FunctionCatalog functionCatalog,
+            CatalogManager catalogManager) {
+        Set<String> jars =
+                resourceManager.getResources().keySet().stream()
                         .map(ResourceUri::getUri)
-                        .toArray(String[]::new);
-        return buildStringArrayResult("jars", jars);
+                        .collect(Collectors.toSet());
+        Set<FunctionIdentifier> identifiers =
+                functionCatalog.getUserDefinedFunctions(
+                        catalogManager.getCurrentCatalog(), catalogManager.getCurrentDatabase());
+        Catalog catalog = catalogManager.getCatalog(catalogManager.getCurrentCatalog()).get();
+        for (FunctionIdentifier identifier : identifiers) {
```

Comment:
It seems it's a little heavy to check the catalog function one by one. I think it's better we register the resource URL into the ResourceManager. But I think we can leave this to the future work.